### PR TITLE
[DPE-7547] Add spread tests to cover all exposed scripts (18/edge)

### PR DESCRIPTION
**Pull request by taurus-forever**
# Issue

Not all exported snap scripts are tested in spread.

# Solution

Add all missing tests + new base ubuntu-26.04

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ backends:
   craft:
     type: craft
     systems:
+      - ubuntu-26.04
```
Comment (marceloneppel) [inline spread.yaml:7]: Also, add it to the matrix from `.github/workflows/ci.yaml` to enable tests?

File: spread/tests/cli_restore/task.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    INSERT INTO test_restore (name) VALUES ('row1'), ('row2');
+  "
+
+  postgresql.dump -Fc -U postgres -h /tmp -d postgres -f /tmp/test_restore.dump
+  PGPASSWORD=pgpass postgresql.dump -Fc -U postgres -h 127.0.0.1 -d postgres -f /tmp/test_restore_tcp.dump
+
+  postgresql.psql -U postgres -h /tmp -c "DROP TABLE test_restore;"
+
+  postgresql.restore -U postgres -h /tmp -d postgres /tmp/test_restore.dump
+  PGPASSWORD=pgpass postgresql.restore -U postgres -h 127.0.0.1 -d postgres /tmp/test_restore.dump 2>/dev/null || true
```
Comment (marceloneppel) [inline spread/tests/cli_restore/task.yaml]: Why are you redirecting the error to `/dev/null` here?

Shouldn't we check for both "row1" and "row2", then also drop the table again before this second restore?

Comment (taurus-forever) [reply spread/tests/cli_restore/task.yaml]: re: `/dev/null`. Nice catch. Local experiments leftover... changes nothing, but ACK to clean.

Improved test and CI/CD accordingly to the request in https://github.com/canonical/postgresql-snap/pull/155/commits/3d522fb58b9c876f6b2be3a4ef5dbd69e989685b

File: spread.yaml (hunk 1 above)
Comment (taurus-forever) [reply spread.yaml:7]: Done in https://github.com/canonical/postgresql-snap/pull/155/commits/3d522fb58b9c876f6b2be3a4ef5dbd69e989685b

Comment (taurus-forever) [reply spread.yaml:7]: It looks like GH runners ubuntu-26.04(-arm) and not fully ready:
https://github.com/canonical/postgresql-snap/actions/runs/26162578916/job/76958969927?pr=155

The GitHub schedule:
https://github.com/actions/runner-images/issues/13964#issuecomment-4360396351

Also, there are NO 26.04 release notes in: https://github.com/actions/runner-images/tree/main/images/ubuntu

Most probable we should revert this part for a ~month.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -51,6 +51,7 @@ jobs:
         base:
           - { codename: jammy, version: 22.04 }
           - { codename: noble, version: 24.04 }
+        # - { codename: resolute, version: 26.04 }
```
Comment (taurus-forever) [inline .github/workflows/ci.yaml:54]: For the history: https://github.com/canonical/postgresql-snap/pull/155#discussion_r3274016031

**Review by marceloneppel — APPROVED** [review]

**Review by dragomirp — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (taurus-forever) [reply .github/workflows/ci.yaml:54]: https://github.com/canonical/postgresql-snap/pull/165
